framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2204,12 +2212,64 @@ public ParameterizedExecutableType methodFromUse(
 
     if (tree.getKind() == Tree.Kind.METHOD_INVOCATION
         && TreeUtils.isMethodInvocation(tree, objectGetClass, processingEnv)) {
-      adaptGetClassReturnTypeToReceiver(methodType, receiverType);
+      adaptGetClassReturnTypeToReceiver(methodType, receiverType, tree);
     }
 
     return new ParameterizedExecutableType(methodType, typeargs);
   }
 
+  /**
+   * Apply capture conversion to the type arguments of a method invocation.
+   *
+   * @param typeVarMapping mapping from type variable to its argument
+   * @param declTypeVar list of type variable declarations
+   * @return a mapping form type variable to its captured type argument
+   */
+  private Map<TypeVariable, AnnotatedTypeMirror> captureMethodTypeArgs(
+      Map<TypeVariable, AnnotatedTypeMirror> typeVarMapping,
+      List<AnnotatedTypeVariable> declTypeVar) {
+    // TODO: This should happen as part of Java 8 inference.
+    // See Issue #979.
+    Map<TypeVariable, AnnotatedTypeVariable> typeParameter = new HashMap<>();
+    for (AnnotatedTypeVariable t : declTypeVar) {
+      typeParameter.put(t.getUnderlyingType(), t);
+    }
+    Map<TypeVariable, AnnotatedTypeMirror> newTypeVarMapping = new HashMap<>();
+    Map<TypeVariable, AnnotatedTypeVariable> capturedTypeMapping = new HashMap<>();
+    for (Map.Entry<TypeVariable, AnnotatedTypeMirror> entry : typeVarMapping.entrySet()) {
+      AnnotatedTypeMirror originalTypeArg = entry.getValue();
+      TypeVariable typeVariable = entry.getKey();
+      if (originalTypeArg.containsUninferredTypeArguments()) {
+        return typeVarMapping;
```

Comment:
Why is this a normal return?  That is, why does this situation not indicate a bug?

Reply:
I'm not sure I understand your question.  Uninferred type arguments are an indication of a bug, 979.  To work around that bug, we ignore uninferred type arguments.

Reply:
Thanks.  A comment to that effect could be useful.